docs/examples/image_binarization/image_binarization.jl

```diff
@@ -0,0 +1,128 @@
+# ---
+# title: Image Binarization
+# author: Ashwani Rathee
+# date: 2021-7-2
+# ---
+
+# ImageBinarization.jl provides several algorithms for binarizing images
+# into background and foreground(bi-level image). In this demonstration, 
+# we'll be exploring where these algorithms could be useful and learn how
+# different algorithms are suited for different types of tasks.
+
+# Suppose a person wants to make an automatic Sudoku solver using Computer 
+# Vision . Before anything else that person needs to import the image, 
+# preprocess it to optimize it for cell content extraction which can then 
+# be used for solving the sudoku. 
+
+# Let's first import packages and then testimage sudoku
+
+using Images, ImageBinarization, TestImages
+using  CoordinateTransformations, Rotations, ImageTransformations
+
+# Original Image
+
+img = testimage("sudoku")
+
+# Grayscale version as binarize! only accepts Grayscale images
+
+img = Gray.(img)
+
+# Let's first rotate our image
+
+trfm = recenter(RotMatrix(pi / 30), center(img));
+imgw = warp(img, trfm)
+
+# Now let's zoom in a bit
+
+imgw = parent(imgw)[65:490, 65:490]
+
+# Now let's binarize an image using the `Sauvola` algorithm
+
+alg = Sauvola();
+img_otsu = binarize(imgw, alg)
+mosaicview(imgw, img_otsu; ncol = 2)
+
+# Now the differences between the binarized and non-binarized images 
+# become apparent. Let's now implement a function to see how different algorithms 
+# perform for this particular problem.
+
+algs = [
+    "AdaptiveThreshold",
+    "Balanced",
+    "Entropy",
+    "Intermodes",
+    "MinimumError",
+    "Moments",
+    "Niblack",
+    "Otsu",
+    "Polysegment",
+    "Sauvola",
+    "UnimodalRosin",
+    "Yen",
+]
+function binarize_methods(img_input, algs)
+    imgs_binarized = Array[]
+    for i in algs
+        alg = getfield(ImageBinarization, Symbol(i))()
+        img_input1 = binarize(img_input, alg)
+        push!(imgs_binarized, img_input1)
+    end
+    return imgs_binarized
+end
```

Comment:
What's the advantage of this compared to

```julia
algs = [AdaptiveThreshold(), Balanced(), Entropy(), Intermodes(),
        MinimumError(), Moments(), Niblack(), Otsu(),
        Polysegment(), Sauvola(), UnimodalRosin(), Yen()]
outs = map(alg->binarize(img, alg), algs)
```
?

Reply:
I didn't know we could make array of functions like this. Suggested version would be more concise then..